src/Internal/Service/OrderService.php

```diff
@@ -170,7 +170,8 @@ public function import_order_data_to_payment_context( int $order_id, PaymentCont
 
 		$context->set_currency( $currency );
 		$context->set_amount( $amount );
-		$context->set_user_id( $user->ID );
+		// In case we don't have user, we are setting user id to be 0 which could cause more harm since we don't have a real user.
+		$context->set_user_id( $user->ID > 0 ? $user->ID : null );
```

Comment:
Now that we have this line, we do not need the `$user = $this->legacy_proxy->call_function( 'wp_get_current_user' )` a few lines above. Seems like it was only there to make sure a `WP_User` instance is available, even if the user is null. Now that we actually allow the code to work with `null`, we can get rid of it.

Reply:
Also, if it is `null`, there is an exception that `set_user_id` of the context requires an integer. Please modify the context method to allow for null users.

Reply:
> Also, if it is `null`, there is an exception that `set_user_id` of the context requires an integer. Please modify the context method to allow for null users.

I am sorry. I haven't committed that file (it was stashed for unknown reason to me). I have committed it now.

Reply:
> Now that we have this line, we do not need the `$user = $this->legacy_proxy->call_function( 'wp_get_current_user' )` a few lines above. Seems like it was only there to make sure a `WP_User` instance is available, even if the user is null. Now that we actually allow the code to work with `null`, we can get rid of it.

Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7640/commits/7677e2b93910ece4f5903976220b9422fa61aae3